ui/src/localization.ts

```diff
@@ -148,3 +153,55 @@ export const loadTranslations = () => {
     exhaustiveCheck(language)
   }, [data?.myUserPreferences.language])
 }
+
+export const SetMapLanguages = (map: mapboxgl.Map) => {
+  if (isNil(map_language)) {
+    map.addControl(new MapboxLanguage({ defaultLanguage: 'en' }))
+    return
+  }
+
+  switch (map_language) {
```

Comment:
We already have all of this information within the `Userpreferences.tsx` called `languagePreferences`. We should extract this and the elements in languagePreferences into a map within this file which is `LanguageTranslation -> languageCode` Although in this case it does appear that the Chinese traditional & simplified do not use the same code as we do within `UserPreferences` (You might need an object rather than one field in the map).

Rewriting it in this way gives us somewhere more central to pick up on language codes and means things don't get missed in the future. A test can also be written to ensure the language selection list length is the same as the map to stop someone doing something strange in the future.